src/Microsoft.IdentityModel.Protocols.WsFederation/WsFederationMessage.cs

```diff
@@ -376,7 +376,7 @@ public string Wct
         /// Gets or sets 'wctx'.
         /// </summary>
         [property: SuppressMessage("Microsoft.Naming", "CA1704")]
-        public string Wctx
+        public string? Wctx
```

Comment:
What about the other properties in this type? Should other ones be made `string?`?

Reply:
What about:

* Wattr
* Wattrptr
* Wauth

Reply:
added and a few others, which seem to be optional in the spec.

Reply:
@jennyf19 it really depends on what the WsFederationMessage is used for.
The spec specifies that wtrealm is required for a RequestSecurityToken message.
For a Pseudonym Request wtrealm is optional.

Worst that can happen is the request will fail.
I don't see what we are gaining by enforcing them to be not null.
Many times these message flow through a call graph where lower levels set values.

Reply:
@eerhardt discussed with @brentschmaltz  and I think we are going to leave this as-is. As Brent points out, some of the properties can be optional or required depending on the context.

Reply:
> some of the properties can be optional or required depending on the context.

Then my recommendation would be to make them all `string?`. This models how they can be used in the API - for example:

```C#
var m = new WsFederationMessage();
m.Wctx.ToString();  // This will result in a NullReferenceException, but the C# compiler won't warn me that `m.Wctx` could be null
```

It's better to make these properties `string?` if there is a chance they are `null`, than to say "some of these strings may be required in some contexts, so none of these strings can ever be null".

Reply:
@brentschmaltz ^

Reply:
We could mark them as nullable.
What i was saying is that we can't make the NOT nullable.

Reply:
@eerhardt and @brentschmaltz marked them as nullable.